### ISSUE TITLE
[package] icu/73.2: Can't build under msvc v143 with C++20

### ISSUE THREAD

**Issue by AshleighAdams** [conan-io/conan-center-index#20876]
### Description

According to ICU, the project should be able to be build with a C++20 compiler, but attempting to build the Conan packages under msvc/v143/C++20 yields the following build error:

```
formattedval_iterimpl.cpp
cl -nologo -DNDEBUG -DU_STATIC_IMPLEMENTATION   -DHAVE_DLOPEN=0 -DU_HAVE_MMAP=0 -DU_HAVE_DIRENT_H=0 -DU_HAVE_POPEN=0 -DU_HAVE_TZNAME=0 -DU_HAVE_STRTOD_L=0 -DU_HAVE_STRING_VIEW=1 -DU_HAVE_WCSCPY=0  -DU_STATIC_IMPLEMENTATION -DU_RELEASE=1 -D_CRT_SECURE_NO_DEPRECATE -I/d/.conan/.conan/data/icu/73.2/_/_/build/bdfcdb69e84dd2faf96987fad1966733454b501d/src/source/i18n -I/d/.conan/.conan/data/icu/73.2/_/_/build/bdfcdb69e84dd2faf96987fad1966733454b501d/src/source/common   -DU_ATTRIBUTE_DEPRECATED= -DWIN32 -DCYGWINMSVC -DU_I18N_IMPLEMENTATION /std:c++20 -MT -O2 -Ob2 -FS   -GF -nologo -EHsc -Zc:wchar_t -utf-8  -c   -Foformattedval_sbimpl.ao /d/.conan/.conan/data/icu/73.2/_/_/build/bdfcdb69e84dd2faf96987fad1966733454b501d/src/source/i18n/formattedval_sbimpl.cpp
D:/.conan/.conan/data/icu/73.2/_/_/build/bdfcdb69e84dd2faf96987fad1966733454b501d/src/source/i18n/fmtable.cpp(59): error C2666: 'icu_73::Measure::operator ==': overloaded functions have similar conversions
D:\.conan\.conan\data\icu\73.2\_\_\build\bdfcdb69e84dd2faf96987fad1966733454b501d\src\source\i18n\unicode/measure.h(90): note: could be 'bool icu_73::Measure::operator ==(const icu_73::UObject &) const'
D:/.conan/.conan/data/icu/73.2/_/_/build/bdfcdb69e84dd2faf96987fad1966733454b501d/src/source/common\unicode/stringpiece.h(326): note: or       'UBool icu_73::operator ==(const icu_73::StringPiece &,const icu_73::StringPiece &)'
D:\.conan\.conan\data\icu\73.2\_\_\build\bdfcdb69e84dd2faf96987fad1966733454b501d\src\source\i18n\unicode/measure.h(90): note: or 'bool icu_73::Measure::operator ==(const icu_73::UObject &) const' [synthesized expression 'y == x']
D:/.conan/.conan/data/icu/73.2/_/_/build/bdfcdb69e84dd2faf96987fad1966733454b501d/src/source/common\unicode/stringpiece.h(326): note: or 'UBool icu_73::operator ==(const icu_73::StringPiece &,const icu_73::StringPiece &)' [synthesized expression 'y == x']
D:/.conan/.conan/data/icu/73.2/_/_/build/bdfcdb69e84dd2faf96987fad1966733454b501d/src/source/i18n/fmtable.cpp(59): note: while trying to match the argument list '(const icu_73::Measure, const icu_73::Measure)
```

### Package and Environment Details

* Package Name/Version: **icu/73.2**
* Operating System+version: **Windows 11**
* Compiler+version: **msvc v143 (VS2022)**
* Docker image: *n/a*
* Conan version: **1.61.0**
* Python version: *n/a, conan embedded*

### Conan profile

[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=static
compiler.runtime_type=Release
compiler.version=193
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
CONAN_DISABLE_STRICT_MODE=1
[conf]
tools.microsoft.msbuild:verbosity=Normal

### Steps to reproduce

`conan install -g txt icu/73.2@ --profile release_143 --build=missing`

Make sure the command is ran in a VS2022 development prompt, else it will fail to find the compiler

### Logs

<details><summary>Click to expand log</summary>

[log.txt](https://github.com/conan-io/conan-center-index/files/13228838/log.txt)

</details>

**Comment by SpaceIm** [comment]
It seems to be a library issue, nothing coming from recipe itself. You should report this bug to https://unicode-org.atlassian.net/jira/software/c/projects/ICU/issues, and see if a patch can be backported.

**Comment by SpaceIm** [comment]
Know issue: https://unicode-org.atlassian.net/browse/ICU-22401

It's fixed in 74.1

**Comment by AshleighAdams** [comment]
Ah, in that case, I'd be more inclined to request a 74.1 build on Conan Center instead. Will close this issue, thanks
